### PR TITLE
preview 7 signalr auth updates - code sample change

### PR DESCRIPTION
**Pull request by bradygaster**
changed code sample based on team discussion

### CODE REVIEW

File: aspnetcore/signalr/authn-and-authz.md [hunk 1]
```diff
@@ -182,53 +182,74 @@ public class ChatHub : Hub
...
+        {
+            options.AddPolicy("DomainRestricted", policy =>
+            {
+                policy.Requirements.Add(new DomainRestrictedRequirement());
+            });
+        });
 }
 ```
 
+> [!NOTE]
```
Comment (scottaddie) [inline aspnetcore/signalr/authn-and-authz.md]:
```suggestion
```

File: aspnetcore/signalr/authn-and-authz.md [hunk 2]
```diff
@@ -182,53 +182,74 @@ public class ChatHub : Hub
 
 SignalR provides a custom resource to authorization handlers when a hub method requires authorization. The resource is an instance of `HubInvocationContext`. The `HubInvocationContext` includes the `HubCallerContext`, the name of the hub method being invoked, and the arguments to the hub method.
 
-The following example shows a custom handler that only allows users with the same name as the hub method to invoke the method.
+Consider the example of a chat room allowing multiple organization sign-in via Azure Active Directory. Anyone with a Microsoft account can sign in to chat, but only members of the owning organization should be able to ban users or view users' chat histories. Furthermore, we might want to restrict certain functionality from certain users. Using the updated features in Preview 7, this is entirely possible. Note how the `DomainRestrictedRequirement` serves as a custom `IAuthorizationRequirement`. Now that the `HubInvocationContext` resource parameter is being passed in, the internal logic can inspect the context in which the Hub is being called and make decisions on allowing the user to execute individual Hub methods.
```
Comment (scottaddie) [inline aspnetcore/signalr/authn-and-authz.md]: If possible, avoid referencing "Preview 7" in the text. Ideally, these doc changes will be published when Preview 7 ships.

Comment (scottaddie) [inline aspnetcore/signalr/authn-and-authz.md]:
```suggestion
Consider the example of a chat room allowing multiple organization sign-in via Azure Active Directory. Anyone with a Microsoft account can sign in to chat, but only members of the owning organization should be able to ban users or view users' chat histories. Furthermore, we might want to restrict certain functionality from certain users. Using the updated features in ASP.NET Core 3.0, this is entirely possible. Note how the `DomainRestrictedRequirement` serves as a custom `IAuthorizationRequirement`. Now that the `HubInvocationContext` resource parameter is being passed in, the internal logic can inspect the context in which the Hub is being called and make decisions on allowing the user to execute individual Hub methods.
```

File: aspnetcore/signalr/authn-and-authz.md [hunk 3]
```diff
@@ -182,53 +182,74 @@ public class ChatHub : Hub
...
 
-    services.AddAuthorization(options =>
-    {
-        options.AddPolicy("NamedMethod", policy =>
-            policy.Requirements.Add(new HubMethodRequirement()));
-    });
-
-    services.AddSingleton<IAuthorizationHandler, HubMethodAuthorizationHandler>();
+    services
+        .AddAuthorization(options =>
```
Comment (scottaddie) [inline aspnetcore/signalr/authn-and-authz.md]: Nit: Move this function call to the line above.

File: aspnetcore/signalr/authn-and-authz.md [hunk 4]
```diff
@@ -182,53 +182,73 @@ public class ChatHub : Hub
...
+        HubInvocationContext resource)
     {
-        if (context.User?.FindFirst(ClaimTypes.NameIdentifier)?.Value == resource.HubMethodName)
+        if(IsUserAllowedToDoThis(resource.HubMethodName, context.User.Identity.Name) && 
+            context.User.Identity.Name.EndsWith("@microsoft.com"))
         {
             context.Succeed(requirement);
         }
-
+
```
Comment (BrennanConroy) [inline aspnetcore/signalr/authn-and-authz.md]:
```suggestion

```

File: aspnetcore/signalr/authn-and-authz.md (hunk 2 above)
Comment (bradygaster) [reply aspnetcore/signalr/authn-and-authz.md]: copypasta err, thanks for catching

File: aspnetcore/signalr/authn-and-authz.md [hunk 5]
```diff
@@ -182,53 +182,73 @@ public class ChatHub : Hub
...
+    IAuthorizationRequirement
 {
     protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
-                                                   HubMethodRequirement requirement,
-                                                   HubInvocationContext resource)
+        DomainRestrictedRequirement requirement, 
+        HubInvocationContext resource)
     {
-        if (context.User?.FindFirst(ClaimTypes.NameIdentifier)?.Value == resource.HubMethodName)
+        if(IsUserAllowedToDoThis(resource.HubMethodName, context.User.Identity.Name) &&
```
Comment (BrennanConroy) [inline aspnetcore/signalr/authn-and-authz.md]:
```suggestion
        if (IsUserAllowedToDoThis(resource.HubMethodName, context.User.Identity.Name) && 
```

File: aspnetcore/signalr/authn-and-authz.md [hunk 6]
```diff
@@ -182,53 +182,73 @@ public class ChatHub : Hub
...
-public class HubMethodRequirement : IAuthorizationRequirement { }
-
-public class ChatHub : Hub
-{
-    [Authorize("NamedMethod")]
-    public void Steve()
+    private bool IsUserAllowedToDoThis(string hubMethodName,
+        string currentUsername)
     {
+        return !(currentUsername.Contains("asdf42@microsoft.com") &&
```
Comment (BrennanConroy) [inline aspnetcore/signalr/authn-and-authz.md]:
```suggestion
        return !(currentUsername.Equals("asdf42@microsoft.com") && 
```

Comment (bradygaster) [reply aspnetcore/signalr/authn-and-authz.md]: dang it i missed that.

**Comment by bradygaster** [comment]
@scottaddie - i'd say merge this once the team approves it, or EOW, whatever comes first. :) It'd be sweet if we could have this change published in prep for the preview 7 drop.

**Pull request merged**